File: iOS/DateMate/UI/Auth/SignIn/SignInUiState.swift
//
//  SignInUiState.swift
//  DateMate
//
//  Created by 홍희표 on 2025/11/18.
//

import Foundation

struct SignInUiState {
    var email: String = ""
    var emailError: String? = nil
    var password: String = ""
    var passwordError: String? = nil
    var isLoading: Bool = false
    var message: String? = nil
    var isSubmitEnabled: Bool {
        let isEmailValid = !email.isEmpty && emailError == nil
        let isPasswordValid = !password.isEmpty && passwordError == nil
        return isEmailValid && isPasswordValid
    }
}

File: Android/app/src/main/java/com/hhp227/datemate/ui/auth/signin/SignInUiState.kt
package com.hhp227.datemate.ui.auth.signin

data class SignInUiState(
    val email: String = "",
    val emailError: String? = null,
    val password: String = "",
    val passwordError: String? = null,
    val isLoading: Boolean = false,
    val message: String? = null
) {
    val isSubmitEnabled: Boolean
        get() {
            val isEmailValid = email.isNotBlank() && emailError == null
            val isPasswordValid = password.isNotBlank() && passwordError == null
            return isEmailValid && isPasswordValid
        }
}